test/NuGet.Core.Tests/NuGet.CommandLine.Xplat.Tests/MSBuildAPIUtilityTests.cs

```diff
@@ -0,0 +1,385 @@
+            ProjectItem packageReference = project.Xml.Items.LastOrDefault(i => i.ItemType == "PackageReference" && i.EvaluatedInclude.Equals("X"));
+            Assert.True(packageReference != null, packageReference.ItemType);
+        }
+
+        [Fact]
+        public void AddPackageReferenceIntoProjectFileWhenItemGroupDoesExist_Success()
+        {
+            // Set up
+            var testDirectory = TestDirectory.Create();
+            var projectCollection = new ProjectCollection(
+                            globalProperties: null,
+                            remoteLoggers: null,
+                            loggers: null,
+                            toolsetDefinitionLocations: ToolsetDefinitionLocations.Default,
+                            // Having more than 1 node spins up multiple msbuild.exe instances to run builds in parallel
+                            // However, these targets complete so quickly that the added overhead makes it take longer
+                            maxNodeCount: 1,
+                            onlyLogCriticalEvents: false,
+                            loadProjectsReadOnly: false);
+
+            var projectOptions = new ProjectOptions
+            {
+                LoadSettings = ProjectLoadSettings.DoNotEvaluateElementsWithFalseCondition,
+                ProjectCollection = projectCollection
+            };
+
+            // Set up project file
+            string projectContent =
+@$"<Project Sdk=""Microsoft.NET.Sdk"">
+<PropertyGroup>                   
+<TargetFramework>net6.0</TargetFramework>
+</PropertyGroup>
+<ItemGroup>
+<PackageReference Include=""Y"" />
+</ItemGroup>
+</Project>";
+            File.WriteAllText(Path.Combine(testDirectory, "projectA.csproj"), projectContent);
+            var project = Project.FromFile(Path.Combine(testDirectory, "projectA.csproj"), projectOptions);
+
+            var msObject = new MSBuildAPIUtility(logger: new TestLogger());
+            // Getting all the item groups in a given project
+            var itemGroups = msObject.GetItemGroups(project);
+            // Getting an existing item group that has package reference(s)
+            var itemGroup = msObject.GetItemGroup(itemGroups, "PackageReference");
+
+            var libraryDependency = new LibraryDependency
+            {
+                LibraryRange = new LibraryRange(
+                        name: "X",
+                        versionRange: VersionRange.Parse("1.0.0"),
+                        typeConstraint: LibraryDependencyTarget.Package)
+            };
+
+            // Act
+            msObject.AddPackageReferenceIntoItemGroupCPM(project, itemGroup, libraryDependency);
+            project.Save();
+
+            // Assert
+            Assert.Contains(@$"<PackageReference Include=""X"" />", File.ReadAllText(Path.Combine(testDirectory, "projectA.csproj")));
+            Assert.DoesNotContain(@$"<Version = ""1.0.0"" />", File.ReadAllText(Path.Combine(testDirectory, "projectA.csproj")));
```

Comment:
I think we can slightly improve the performance by reading text from the project only once.

Reply:
Agreed, will fix that.